gap/greens/froidure-pin.gi

```diff
@@ -606,39 +606,47 @@ function(S)
 end);
 
 InstallMethod(LeftGreensMultiplierNC,
-"for a semigroup that can use Froidure-Pin and L-related elements with one",
+"for a semigroup that can use Froidure-Pin and L-related elements",
 [IsSemigroup and CanUseFroidurePin,
- IsMultiplicativeElementWithOne,
- IsMultiplicativeElementWithOne],
+ IsMultiplicativeElement,
+ IsMultiplicativeElement],
 function(S, a, b)
-  local gens, D, path;
+  local gens, D, path, a1, b1;
   gens := GeneratorsOfSemigroup(S);
   D := LeftCayleyDigraph(S);
-  a := PositionCanonical(S, a);
-  b := PositionCanonical(S, b);
-  path := NextIterator(IteratorOfPaths(D, a, b));
+  a1 := PositionCanonical(S, a);
+  b1 := PositionCanonical(S, b);
+  path := NextIterator(IteratorOfPaths(D, a1, b1));
   if path = fail then
     # This can occur when, for example, a = b and S is not a monoid.
-    return One(gens);
+    if IsMultiplicativeElement(a) and IsMultiplicativeElement(b) then
```

Comment:
I don't think this does exactly what you want it to, maybe you meant `not IsMultiplicativeElementWithOne(a)` and similar for `b`?

Reply:
Still missing, and the documentation for Left/RightGreensMultiplier should be updated to include:

> If <A>S</A> is a semigroup but not a monoid, then the return value is either of the same type as the elements of <A>S</A> or an adjoined identity.

Reply:
Missing "WithOne" fixed. The documentation is possibly more delicate than that, as an fp semigroup can be a monoid but GAP cannot recognise it as one. One example is < s1, s2 | s1^2=s1, s2^2=s2, s1*s2=s2, s2*s1=s2>. s1 is a "one" but GAP does not know that.